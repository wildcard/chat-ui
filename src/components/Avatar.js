import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { withStyles } from 'material-ui/styles';
import Avatar from 'material-ui/Avatar';
import { chooseAvatar } from '../chat-avatar';

const styles = {
  avatar: {
    margin: 10,
  },
  tiny: {
    width: 20,
    height: 20,
  },
  big: {
    width: 60,
    height: 60,
  },
};

function ChatAvatar(props) {
  const { classes, username, src, tiny, big, ...other } = props;

  const avatarImageSrc = src || chooseAvatar(username);

  return (
    <Avatar
      alt={username}
      src={avatarImageSrc}
      className={classNames(
        {
          [classes.big]: big,
          [classes.tiny]: tiny,
        },
        classes.avatar
      )}
      {...other}
    />
  );
}

ChatAvatar.propTypes = {
  classes: PropTypes.object.isRequired,
  username: PropTypes.string.isRequired,
  src: PropTypes.string,
  big: PropTypes.bool,
  tiny: PropTypes.bool,
};

export default withStyles(styles)(ChatAvatar);
