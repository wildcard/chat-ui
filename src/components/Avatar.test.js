import React from 'react';
import Avatar from './Avatar';
import renderer from 'react-test-renderer';

describe('snapshot', () => {
  it('', () => {
    const component = renderer.create(
      <Avatar username="kobi.kadosh"/>,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('tiny', () => {
    const component = renderer.create(
      <Avatar tiny username="kobi.kadosh"/>,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('big', () => {
    const component = renderer.create(
      <Avatar big username="kobi.kadosh"/>,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('src', () => {
    const component = renderer.create(
      <Avatar username="kobi.kadosh" src="https://s3.amazonaws.com/uifaces/faces/twitter/jeremiespoken/128.jpg"/>,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
