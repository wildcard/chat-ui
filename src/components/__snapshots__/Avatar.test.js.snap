// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot  1`] = `
<div
  className="MuiAvatar-root-4 ChatAvatar-avatar-1"
>
  <img
    alt="kobi.kadosh"
    className="MuiAvatar-img-6"
    sizes={undefined}
    src="https://spotim-demo-chat-server.herokuapp.com/avatars/003-pikachu.png"
    srcSet={undefined}
  />
</div>
`;

exports[`snapshot big 1`] = `
<div
  className="MuiAvatar-root-4 ChatAvatar-big-3 ChatAvatar-avatar-1"
>
  <img
    alt="kobi.kadosh"
    className="MuiAvatar-img-6"
    sizes={undefined}
    src="https://spotim-demo-chat-server.herokuapp.com/avatars/003-pikachu.png"
    srcSet={undefined}
  />
</div>
`;

exports[`snapshot src 1`] = `
<div
  className="MuiAvatar-root-4 ChatAvatar-avatar-1"
>
  <img
    alt="kobi.kadosh"
    className="MuiAvatar-img-6"
    sizes={undefined}
    src="https://s3.amazonaws.com/uifaces/faces/twitter/jeremiespoken/128.jpg"
    srcSet={undefined}
  />
</div>
`;

exports[`snapshot tiny 1`] = `
<div
  className="MuiAvatar-root-4 ChatAvatar-tiny-2 ChatAvatar-avatar-1"
>
  <img
    alt="kobi.kadosh"
    className="MuiAvatar-img-6"
    sizes={undefined}
    src="https://spotim-demo-chat-server.herokuapp.com/avatars/003-pikachu.png"
    srcSet={undefined}
  />
</div>
`;
